lib/iris/fileformats/_nc_load_rules/helpers.py

```diff
@@ -1024,6 +1032,57 @@ def reorder_bounds_data(bounds_data, cf_bounds_var, cf_coord_var):
     return bounds_data
 
 
+################################################################################
+def _normalise_bounds_units(
+    points_units: str, cf_bounds_var: CFBoundaryVariable, bounds_data: ArrayLike
+) -> Optional[MaskedArray]:
+    """Ensure bounds have units compatible with points.
+
+    If required, the `bounds_data` will be converted to the `points_units`.
+    If the bounds units are not convertible, a warning will be issued and
+    the `bounds_data` will be ignored.
+
+    Bounds with invalid units will be gracefully left unconverted and passed through.
+
+    Parameters
+    ----------
+    points_units : str
+        The units of the coordinate points.
+    cf_bounds_var : CFBoundaryVariable
+        The serialized NetCDF bounds variable.
+    bounds_data : MaskedArray
+        The pre-processed data of the bounds variable.
+
+    Returns
+    -------
+    MaskedArray or None
+        The bounds data with the same units as the points, or ``None``
+        if the bounds units are not convertible to the points units.
+
+    """
+    bounds_units = get_attr_units(cf_bounds_var, {})
```

Comment:
In the case where the bounds have an invalid unit, this will raise an error with the message:
"Ignoring netCDF variable {!r} invalid units {!r}"

It strikes me that this message might be a tad unclear about if it is ignoring the variable or if it is ignoring the units. I feel like there could be a case for making this warning clearer similar to the warning below. e.g. :
"Ignoring invalid units {units} on NetCDF variable {var}."

Reply:
Happy to change this warning to help clarify :+1: